lib/library.rb

```diff
@@ -0,0 +1,57 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def add_author(author_obj)
+    @authors << author_obj
+  end
+
+  def add_book(book_obj)
```

Comment:
If you're going to have this as it's own method, you'll need to make sure it's called from the add_author method so it gets called for every book every time an author is added.